### PR TITLE
Port all samples to be compatible with Zephyr changes

### PR DESCRIPTION
**Pull request by bigdinotech**
-port all samples to use new CDC-ACM driver
-change SRAM memory mapping to increase SRAM allocated to ARC

### CODE REVIEW

**Comment by bigdinotech** [comment]
@eriknyquist @SidLeung ready for review

File: common/arduino101_services/arduino101_main.c [hunk 1]
```diff
@@ -26,21 +26,73 @@
...
+void copy_device_descriptor(char *dest, uint8_t *source, const char *default_value, uint8_t len, uint8_t max)
+{
+	if ((len > 0) && (len < max))
+	{
+		memcpy(dest, source, len);
+		dest[len] = '\0';
+	}
+	else
+	{
+		memcpy(dest, default_value, strlen(default_value));
```
Comment (eriknyquist) [inline common/arduino101_services/arduino101_main.c:58]: So if the passed string is too long, it uses default values instead? That's a bit weird. I think most people would expect their long string to be truncated to fit.

**Comment by bigdinotech** [comment]
len is read from the OTP so if it is beyond MAX value it's value is most likely 0xFF meaning OTP has not been flashed or improperly flashed. We could truncate it, but we would be truncating garbage.

**Comment by eriknyquist** [comment]
Ah, I see. OK.

**Review by eriknyquist — APPROVED** [review]

**Pull request merged**